common/scripts/changelog/check.mjs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Script to check that changefile has been generated for each Pull Request.
+ *
+ * This script is only useful in the context of PR validation.
+ *
+ * Usage:
+ *   node common/scripts/changelog/check.mjs origin/main origin/feature-pony
+ */
+
+import { REPO_ROOT } from "../lib/constants.mjs";
+import { exec_output } from "../lib/exec.mjs";
+import path from 'path';
+import { parseNewChangeFiles } from "./utils.mjs";
+
+async function main() {
+  const [base, head] = parseArgs(process.argv);
+  const gitLogStdout = await exec_output(`git log --name-status ${base}..${head} -- ${path.join(REPO_ROOT, 'change/')}`);
+  const newChangeFiles = parseNewChangeFiles(gitLogStdout);
+  if (!newChangeFiles?.length) {
+    console.error('No changefile detected! Please run `rush changelog` to document your change.');
```

Comment:
Not sure if I missed some part here or not, we might want to check if the change is happening under packages/ folder then asking for a changelog, for those changes in storybook, samples, no changelogs required

Reply:
So... I stayed dumb in this area and we do now require a changelog for _any_ PR.

If this becomes a problem we can revisit adding a check for the actual changes. In my experience, asking for a change-file in each PR is not onerous, and there's always the `none` type for some trivial PRs.